libs/client-api/src/native/http_native.rs

```diff
@@ -122,24 +121,43 @@ impl Client {
   }
 
   async fn inner_refresh_token(&self) -> Result<(), AppResponseError> {
-    let retry_strategy = FixedInterval::new(Duration::from_secs(10)).take(4);
-    let action = RefreshTokenAction::new(self.token.clone(), self.gotrue_client.clone());
-    match RetryIf::spawn(retry_strategy, action, RefreshTokenRetryCondition).await {
-      Ok(_) => {
-        event!(tracing::Level::INFO, "refresh token success");
-        Ok(())
-      },
-      Err(err) => {
-        let err = AppError::from(err);
-        event!(tracing::Level::ERROR, "refresh token failed: {}", err);
+    let mut current_retry_count = 0;
+    let max_retry_count = 4;
 
-        // If the error is an OAuth error, unset the token.
-        if err.is_unauthorized() {
-          self.token.write().unset();
-        }
-        Err(err.into())
-      },
+    let refresh_token = self
```

Comment:
I favor using the retry action because it is self-contained and allows for straightforward configuration adjustments, such as changing the strategy.

Reply:
I'll try to see if I can integrate it back into it